src/main/java/com/microsoft/azure/relay/ClientWebSocket.java

```diff
@@ -344,6 +341,15 @@ public void onError(Session session, Throwable cause) {
 		RelayLogger.throwingException(cause, this);
 	}
 	
+	private static ByteBuffer deepCopyByteBuffer(ByteBuffer original) {
+		ByteBuffer clone = ByteBuffer.allocate(original.capacity());
+		clone.put(original.array());
```

Comment:
>riginal.array() [](start = 13, length = 15)

Just curious, did you consider not copying the insignificant bytes?

Reply:
I was just about to leave a comment, in that case I would probably even omit this deepcopy method and just only extract the array segment that matters. Had this here for now mostly for the method name.

---
In reply to: [296936087](https://github.com/Azure/azure-relay-java/pull/48#discussion_r296936087) [](ancestors = 296936087)